File: Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Customers/Commands/WithdrawFunds.cs
namespace Tuckshop.Core.Models.Customers.Commands
{
  using Neo.Model;
  using Neo.Model.Validation;
  using System.ComponentModel.DataAnnotations;
  using System.ComponentModel.DataAnnotations.Schema;

  /// <summary>
  /// Command to withdraw funds from a customer's wallet
  /// </summary>
  public class WithdrawFunds : ModelBase<WithdrawFunds>
  {
    /// <summary>
    /// Gets or sets the Customer Id.
    /// </summary>
    [Required]
    public int CustomerId { get; set; }

    /// <summary>
    /// Gets or sets the amount to withdraw
    /// </summary>
    [Required]
    [Column(TypeName = "money")]
    public decimal Amount { get; set; }

    /// <summary>
    /// Gets or sets the reason for withdrawal (optional, for admin auditing)
    /// </summary>
    [MaxLength(255)]
    public string Reason { get; set; } = string.Empty;

    /// <inheritdoc/>
    protected override void AddBusinessRules(ValidationRules<WithdrawFunds> rules)
    {
      base.AddBusinessRules(rules);
      rules.FailWhen(c => c.Amount <= 0, "Amount must be greater than zero");
    }
  }
}

File: Client/Tuckshop.WebApp/src/Domain/Models/Customers/Commands/WithdrawFunds.ts
import { Attributes, ModelBase, Rules, Validation } from '@singularsystems/neo-core';

export default class WithdrawFunds extends ModelBase {
    static typeName = "WithdrawFunds";

    constructor() {
        super();
        this.makeObservable();
    }

    @Rules.Required()
    public customerId: number = 0;

    @Attributes.Float()
    public amount: number = 0;

    @Rules.StringLength(255)
    public reason: string = "";

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<WithdrawFunds>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew || !this.reason) {
            return "New withdraw funds";
        } else {
            return this.reason;
        }
    }
}